source/Halibut/Transport/Protocol/ITypeRegistry.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Halibut.Transport.Protocol
+{
+    public interface ITypeRegistry
+    {
+        void Register(params Type[] registeredServiceTypes);
+        void RegisterType(Type type, string path, bool ignoreObject);
+        bool IsInAllowedTypes(Type type);
+        void AddToMessageContract(params Type[] types);
```

Comment:
I wonder if the `AddToMessageContract` and `Register` methods should be combined (and named appropriately).

`AddToMessageContract` is just a wrapper over `Register`. The only difference is that it locks `messageContractTypes` first, and only registers types that are not in the `messageContractTypes` list already.

The `messageContractTypes` list isn't used for anything else, so it's just an optimisation.

I can't help but wonder if we simply make `Register` do the checks that `AddToMessageContract` does. 

It won't affect the way the system behaves contractually. But it will be less confusing for future developers.